official/projects/lra/linformer_experiments.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Linformer experiments."""
+# pylint: disable=g-doc-return-or-yield,line-too-long
+import dataclasses
+
+from official.core import config_definitions as cfg
+from official.core import exp_factory
+from official.modeling import optimization
+from official.nlp.configs import encoders
+from official.nlp.data import sentence_prediction_dataloader
+from official.nlp.tasks import sentence_prediction
+from official.projects.lra.linformer import LinformerEncoderConfig
+
+from official.projects.lra import lra_dual_encoder_dataloader
+from official.projects.lra import lra_dual_encoder_task
+
+AdamWeightDecay = optimization.AdamWeightDecayConfig
+PolynomialLr = optimization.PolynomialLrConfig
+PolynomialWarmupConfig = optimization.PolynomialWarmupConfig
+
+
+@dataclasses.dataclass
```

Comment:
I don't feel we need to subclass.
Could you update according to this example?
https://github.com/tensorflow/models/blob/master/official/nlp/configs/pretraining_experiments.py#L25

Reply:
Updated this script and others accordingly